evalml/pipelines/time_series_pipeline_base.py

```diff
@@ -151,7 +151,9 @@ def _drop_time_index(self, X, y):
         if self.should_drop_time_index and self.time_index in X.columns:
             X_schema = X.ww.schema
             y_schema = y.ww.schema
+            index_name = X.index.name
```

Comment:
Needed to preserve the index name of the index here.  The resulting error I was seeing was complaining about the index of X being renamed to "date" when "date" was also a column, so the subsequent access of X[self.time_index] was coming up as ambiguous.  All these KeyError(0)'s [here](https://git.alteryx.com/tempo2/tempo2-server/-/jobs/31026234) are that manifestation of that error that is fixed by these two lines.